### PR TITLE
Improve code quality in MachineLearning library

### PR DESCRIPTION
**Pull request by msoeken**

### CODE REVIEW

File: MachineLearning/src/GradientEstimation.qs [hunk 1]
```diff
@@ -12,20 +12,15 @@ namespace Microsoft.Quantum.MachineLearning {
...
-    internal operation _EstimateDerivativeWithParameterShift(
+    internal operation EstimateDerivativeWithParameterShift(
         inputEncoder : StateGenerator,
         model : SequentialModel,
         parameters : (Double[], Double[]),
         nQubits : Int,
         nMeasurements : Int
     ) : Double {
         return EstimateRealOverlapBetweenStates(
-            _ApplyLEOperationToRawRegister(inputEncoder::Prepare, _),
```
Comment (cgranade) [inline MachineLearning/src/GradientEstimation.qs:28]: We should test to make sure that the library still works when used from IQ#, given https://github.com/microsoft/iqsharp/issues/606.

File: MachineLearning/tests/StructureTests.qs [hunk 2]
```diff
@@ -13,7 +13,7 @@ namespace Microsoft.Quantum.MachineLearning.Tests {
         let model = Default<ML.SequentialModel>()
             w/ Structure <- [
                 ML.ControlledRotation((3, [7, 9]), PauliX, 0),
-                ML.ControlledRotation((8, new Int[0]), PauliY, 1)
+                ML.ControlledRotation((8, [0]), PauliY, 1)
```
Comment (cgranade) [inline MachineLearning/tests/StructureTests.qs:16]:
```suggestion
                ML.ControlledRotation((8, []), PauliY, 1)
```

**Review by cgranade — APPROVED** [review]
Nice to highlight how recent Q# features really help improve code quality!

File: MachineLearning/src/GradientEstimation.qs (hunk 1 above)
Comment (tcNickolas) [reply MachineLearning/src/GradientEstimation.qs:28]: Most likely it won't, I ran into this when trying to update the Katas - even if the lambda is in the "backend" and not in the notebook itself, compilation fails

File: MachineLearning/tests/StructureTests.qs (hunk 2 above)
Comment (msoeken) [reply MachineLearning/tests/StructureTests.qs:16]: Good catch!

**Pull request merged**